# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: parser/html/nsHtml5TreeOpExecutor.h
# Commit: 77388b0795b4
# Full Hash: 77388b0795b4f9a30171c922d6b649c52e5f6c87
# Author: Henri Sivonen <hsivonen@hsivonen.fi>
# Date: 2021-12-23 09:34:36
# Description:
#   Bug 1745142 - Communicate encoding commitment via speculative load queue. r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D133996
# ==============================================================================

diff -r a388127095db -r 77388b0795b4 parser/html/nsHtml5TreeOpExecutor.h
--- a/parser/html/nsHtml5TreeOpExecutor.h	Wed Dec 22 15:47:08 2021 +0000
+++ b/parser/html/nsHtml5TreeOpExecutor.h	Wed Dec 22 15:54:49 2021 +0000
@@ -22,6 +22,7 @@
 #include "nsHashKeys.h"
 #include "mozilla/LinkedList.h"
 #include "nsHtml5DocumentBuilder.h"
+#include "nsCharsetSource.h"
 
 class nsHtml5Parser;
 class nsHtml5StreamParser;
@@ -60,21 +61,6 @@
    */
   bool mSuppressEOF;
 
-  /**
-   * Set to true if CommitToInternalEncoding() was unable to flush all
-   * pending operations. In that case, this flag signals to
-   * ContinueInterruptedParsingAsync() that another attempt to call
-   * CommitToInternalEncoding() (from the event loop) is need.
-   *
-   * This may happen in particular when extensions inject scripts
-   * to the document, and the script injection prevents
-   * CommitToInternalEncoding() from completing in one attempt.
-   *
-   * This can also happen as a result of the flush being slow enough
-   * that the flush is interrupted based on time.
-   */
-  bool mPendingEncodingCommitment;
-
   bool mReadingFromStage;
   nsTArray<nsHtml5TreeOperation> mOpQueue;
   nsHtml5StreamParser* mStreamParser;
@@ -194,16 +180,21 @@
 
   void RunFlushLoop();
 
-  void RunFlushLoopOrCommitToInternalEncoding();
-
   nsresult FlushDocumentWrite();
 
   void CommitToInternalEncoding();
 
+  void TakeOpsFromStage();
+
   void MaybeSuspend();
 
   void Start();
 
+  void SetDocumentCharsetAndSource(NotNull<const Encoding*> aEncoding,
+                                   nsCharsetSource aCharsetSource);
+
+  void UpdateCharsetSource(nsCharsetSource aCharsetSource);
+
   void NeedsCharsetSwitchTo(NotNull<const Encoding*> aEncoding, int32_t aSource,
                             uint32_t aLineNumber);
 